### ISSUE TITLE
DL3009 Warning when clean done as a later statement?

### ISSUE THREAD

**Issue by mtearle** [hadolint/hadolint#311]
- [x] This is a bug report
- [ ] This is a feature request
- [X] I searched existing issues before opening this one

### Expected behavior
RUN apt-get clean \
  && rm -rf /var/lib/apt/lists/*

(no warnings)

### Actual behavior

/dev/stdin:9 DL3009 Delete the apt-get lists after installing something

### Steps to reproduce the behavior

In Dockerfile....

Several throughout ...

RUN apt-get install --no-install-recommends -y &lt;package&gt;

Final cleanup ...

RUN apt-get clean \
  && rm -rf /var/lib/apt/lists/*

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
$ docker run --rm -i hadolint/hadolint hadolint --version
Haskell Dockerfile Linter v1.16.2-0-g3b6f5a6
```

**Dockerfile** (if relevant)

```Dockerfile
(paste content of your Dockerfile)
```

**Additional environment details (OS, stack version, etc.)**

**Comment by mtearle** [comment]
#63 and #185 possibly related?

**Comment by varnav** [comment]
Same problem. This is what I have:

`RUN DEBIAN_FRONTEND=noninteractive apt-get update && apt-get -y install --no-install-recommends openssl ca-certificates wget
RUN wget -q https://github.com/EOSIO/eos/releases/download/v1.7.2/eosio_1.7.2-1-ubuntu-18.04_amd64.deb
RUN DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends ./eosio_1.7.2-1-ubuntu-18.04_amd64.deb
RUN apt-get clean && rm -rf /var/lib/apt/lists/*`

And this is what I get:

Dockerfile:6 DL3009 Delete the apt-get lists after installing something

**Comment by lorenzo** [comment]
That is expected, it should be done in the same step to reduce the size of the image. You can ignore the rule if you don’t care about the size.

**Comment by varnav** [comment]
Maybe we need to add the explanation here:

https://github.com/hadolint/hadolint/wiki/DL3009

Or people will keep asking.

**Comment by lorenzo** [comment]
@varnav would you be kind to add a note about it in the wiki?

**Comment by varnav** [comment]
Done!
